app/Services/Allegro/AllegroOfferCheckerAdapter.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Services\Allegro;
+
+use App\Enums\AvailableStore;
+use App\Models\Offer;
+use App\Services\AbstractOfferCheckerAdapter;
+use App\Services\Contract\OfferCheckerInterface;
+use App\Services\UrlService;
+use App\Traits\PriceComparisonTrait;
+
+class AllegroOfferCheckerAdapter extends AbstractOfferCheckerAdapter implements OfferCheckerInterface
+{
+    use PriceComparisonTrait;
+    public const DOMAIN = 'allegro.pl';
+    public AllegroService $service;
+    public function __construct($offer)
+    {
+        $this->service = new AllegroService($offer);
+    }
+    public function getOfferPrice(): float
+    {
+        return $this->service->getOfferPrice();
```

Comment:
Zastanawiam się czy taki osobny serwis który tak na prawdę robi robotę jest tutaj potrzeby. Może to byłby jakiś Allegro connetor który odpowiadałby tylko za logowanie? Bo teraz to w allegro service jest cała logia, która tak naprawdę mogłaby być tutaj

Reply:
Fakt, ale to wtedy już nie będzie adapterem? I cały schemat się rozwali? Dodatkowo, wydaje mi się, że to dobra droga "na przyszłość", gdybym chciał żeby zachowania dla jednego i drugiego serwisu się trochę różniły (np. ze względu na kategorię oferty)

Reply:
No może właśnie nie powinien być adapterem chyba machnąłem się z nazwą. To chyba bardziej wzorzec strategia. chodzi o to że właśnie nie bardzo powinno się robić metody które tylko używają innych metod, rzadko jest to dobrym pomysłem bo to komplikuje kod, bo metoda z adaptera może być użyta 5 razy a metoda z serwisu raz, albo odwrotnie i nie wiesz gdzie jest to zaimplementowane. Ale jeśli uważasz że to słuszne podejście to może tak zostać. Tutaj to bardziej podchodzi pod strategy pattern a nie adapter

Reply:
Ok, rozumiem - zobaczę w jaką stronę to pójdzie w takim razie i jeśli nic się nie zmieni zastosuję strategy pattern